src/utils/content-to-jsx.test.tsx

```diff
@@ -0,0 +1,131 @@
+  });
+
+  it('generates the expected html when passed a Heading', () => {
+    const result = contentToJsx({
+      type: 'Heading', depth: 1, content: 'heading', id: 'h1',
+    });
+
+    expect(result).toStrictEqual(<Heading id={'h1'} content={'heading'} headingLevel={1}/>);
+  });
+
+  it('generates the expected html when passed a Cite', () => {
+    const result = contentToJsx({
+      type: 'Cite',
+      content: 'I am a citation',
+      target: 'target',
+    });
+
+    expect(result).toStrictEqual(<a href={'target'}>I am a citation</a>);
+  });
+
+  it('generates the expected html when passed a Link', () => {
+    const result = contentToJsx({
+      type: 'Link',
+      content: 'I am a link',
+      target: 'target',
+    });
+
+    expect(result).toStrictEqual(<a href={'target'}>I am a link</a>);
+  });
+
+  it('generates the expected html when passed a Paragraph', () => {
+    const result = contentToJsx({
+      type: 'Paragraph',
+      content: 'I am a paragraph',
+    });
+
+    expect(result).toStrictEqual(<p>I am a paragraph</p>);
+  });
+
+  it('generates the expected html when passed a Emphasis', () => {
+    const result = contentToJsx({
+      type: 'Emphasis',
+      content: 'I am emphasised',
+    });
+
+    expect(result).toStrictEqual(<em>I am emphasised</em>);
+  });
+
+  it('generates the expected html when passed a Strong', () => {
+    const result = contentToJsx({
+      type: 'Strong',
+      content: 'I am strong',
+    });
+
+    expect(result).toStrictEqual(<strong>I am strong</strong>);
+  });
+
+  it('generates the expected html when passed a Superscript', () => {
+    const result = contentToJsx({
+      type: 'Superscript',
+      content: 'I am super',
+    });
+
+    expect(result).toStrictEqual(<sup>I am super</sup>);
+  });
+
+  it('generates the expected html when passed a Subscript', () => {
+    const result = contentToJsx({
+      type: 'Subscript',
+      content: 'I am a subscript',
+    });
+
+    expect(result).toStrictEqual(<sub>I am a subscript</sub>);
+  });
+
+  it('generates the expected html when passed a Date', () => {
+    const result = contentToJsx({
+      type: 'Date',
+      content: '13/01/2001',
+    });
+
+    expect(result).toStrictEqual(<time>13/01/2001</time>);
+  });
+
+  it('generates the expected html when passed a Figure', () => {
+    const result = contentToJsx({
+      type: 'Figure',
+      content: 'I am a figure',
+      caption: 'I am a caption',
+      label: 'I am a label',
+      id: 'id',
+    });
+
+    expect(result).toStrictEqual(
+      <figure>
+        <label>I am a label</label>
+        I am a figure
+        <figcaption>I am a caption</figcaption>
+      </figure>,
+    );
+  });
+
+  it('generates the expected html when passed a ImageObject', () => {
+    const result = contentToJsx({
+      type: 'ImageObject',
+      contentUrl: 'https://placekitten.com/500/300',
+      content: [],
```

Comment:
I'm guessing you've added this to satisfy the type. Would it be better to alter `ImageObjectContent` so it's not a union with `DecoratedContent` ?

Reply:
Actually, following the [Stencila Schema definition for `ImageObject`](https://github.com/stencila/schema/blob/master/ts/types.ts#L1088), I think only contentUrl is required, content can be there, but should be optional.

The type definitions in the link above are actually quite helpful in understanding encoda's JSON output.